data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginModel.java

```diff
@@ -0,0 +1,103 @@
+package com.amazon.dataprepper.model.configuration;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Model class for a Plugin in Configuration YAML containing name of the Plugin and its associated settings
+ *
+ * @since 1.2
+ */
+@JsonSerialize(using = PluginModel.PluginModelSerializer.class)
+@JsonDeserialize(using = PluginModel.PluginModelDeserializer.class)
+public class PluginModel {
+
+    private final String pluginName;
+    private final Map<String, Object> pluginSettings;
+
+    public PluginModel(final String pluginName, final Map<String, Object> pluginSettings) {
+        this.pluginName = pluginName;
+        this.pluginSettings = pluginSettings;
+    }
+
+    public String getPluginName() {
+        return pluginName;
+    }
+
+    public Map<String, Object> getPluginSettings() {
+        return pluginSettings;
+    }
+
+    /**
+     * Custom Serializer for Plugin Model
+     *
+     * @since 1.2
+     */
+    static class PluginModelSerializer extends StdSerializer<PluginModel> {
+
+        private final ObjectMapper mapper = new ObjectMapper();
+
+        public PluginModelSerializer() {
+            this(null);
+        }
+
+        public PluginModelSerializer(Class<PluginModel> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(
+                PluginModel value, JsonGenerator gen, SerializerProvider provider) throws IOException {
```

Comment:
Let's assign variables and parameters that shouldn't be reassigned as final.

There are other variables and parameters in this class and subclasses that should be set to final as well.